libnavui-status/src/main/java/com/mapbox/navigation/ui/status/view/MapboxStatusView.kt

```diff
@@ -0,0 +1,233 @@
+package com.mapbox.navigation.ui.status.view
+
+import android.animation.Animator
+import android.animation.AnimatorInflater
+import android.animation.AnimatorSet
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import android.widget.ProgressBar
+import androidx.annotation.AnimatorRes
+import androidx.annotation.UiThread
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.animation.doOnStart
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import com.mapbox.navigation.ui.status.R
+import com.mapbox.navigation.ui.status.databinding.MapboxStatusViewLayoutBinding
+import com.mapbox.navigation.ui.status.internal.extensions.doOnFinish
+import com.mapbox.navigation.ui.status.model.Status
+
+/**
+ * View for rendering [Status] information.
+ */
+@Suppress("MemberVisibilityCanBePrivate")
+class MapboxStatusView : FrameLayout {
```

Comment:
True, it does look redundant but making this view a FrameLayout instead of ConstraintLayout hides implementation details  and prevents developers from misusing it as parent for their views. (FrameLayout is designed to display only a single item).